app/src/main/java/com/vodafone/v2x/android/hellov2xworld/mapdrawing/CAMDrawing.java

```diff
@@ -18,33 +23,26 @@
  * It also uses {@link ITSDrawing} to get the current map bearing to rotate the CAM icon accordingly.
  */
 public class CAMDrawing {
-
-    private static final String TAG = "CAMDrawing";
+    private final MapView mapView;
     private final IconsFactory iconsFactory;
     private final ITSDrawing itsDrawing;
+    private final HashMap<Long, Marker> listCAMMarkers = new HashMap<>();
 
-    private final Marker[] mCamMarkers = new Marker[50];
+    private static final int maxNumberOfCAM = 50;
     private List<CAMRecord> camRecords = null;
 
-    /**
-     * Constructor of the class CAMDrawing.
-     * It creates an array of markers for displaying CAM information on the map and adds them to the map overlay.
-     * It also sets the default CAM icon for each marker.
-     *
-     * @param map        The map view object to display CAM on the map.
-     * @param itsDrawing The ITSDrawing object to get the current map bearing.
-     */
-    public CAMDrawing(MapView map, ITSDrawing itsDrawing) {
+//    /**
+//     * Constructor of the class CAMDrawing.
+//     * It creates an array of markers for displaying CAM information on the map and adds them to the map overlay.
+//     * It also sets the default CAM icon for each marker.
+//     *
+//     * @param map        The map view object to display CAM on the map.
+//     * @param itsDrawing The ITSDrawing object to get the current map bearing.
+//     */
```

Comment:
Why these lines are commented, instead change the description according to the changes you did